internal/k8s/reconciler/deployer.go

```diff
@@ -44,16 +44,16 @@ func NewDeployer(config DeployerConfig) *GatewayDeployer {
 	}
 }
 
-func (d *GatewayDeployer) Deploy(ctx context.Context, namespace string, config apigwv1alpha1.GatewayClassConfig, gateway *gwv1beta1.Gateway) error {
-	if err := d.ensureServiceAccount(ctx, config, gateway); err != nil {
+func (d *GatewayDeployer) Deploy(ctx context.Context, gateway *K8sGateway) error {
```

Comment:
Simplified to accept a `K8sGateway` which contains all of the individual parameters that were previously accepted